src/megatron/bridge/recipes/qwen_vl/qwen25_vl_step.py

```diff
@@ -0,0 +1,233 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from functools import partial
+from typing import Iterable
+
+import torch
+from megatron.core import parallel_state
+from megatron.core.models.gpt import GPTModel
+from megatron.core.utils import get_batch_on_this_cp_rank, get_model_config
+
+from megatron.bridge.training.config import ConfigContainer
+from megatron.bridge.training.gpt_step import (
+    _create_loss_function,
+    get_packed_seq_params,
+)
```

Comment:
we can spilt these functions out of `gpt_step` to indicate they are more generic

Reply:
i feel gpt_step is the general one lol

Reply:
moved them out to utils